client/state/data-layer/wpcom/sites/comments/index.js

```diff
@@ -58,21 +58,26 @@ const changeCommentStatus = ( { dispatch, getState }, action ) => {
 	);
 };
 
-export const handleChangeCommentStatusSuccess = (
-	{ dispatch },
-	{ commentId, refreshCommentListQuery }
-) => {
+export const handleChangeCommentStatusSuccess = ( { dispatch }, action ) => {
+	const { commentId, refreshCommentListQuery: query, siteId } = action;
 	dispatch( removeNotice( `comment-notice-error-${ commentId }` ) );
-	if ( !! refreshCommentListQuery ) {
-		dispatch( requestCommentsList( refreshCommentListQuery ) );
+	if ( !! query ) {
+		dispatch( requestCommentsList( query ) );
+		if ( !! query.progressId ) {
+			dispatch( updateCommentsProgress( siteId, query.progressId ) );
+		}
 	}
 };
 
-const announceStatusChangeFailure = ( { dispatch }, action ) => {
-	const { commentId, status, previousStatus } = action;
+export const announceStatusChangeFailure = ( { dispatch }, action ) => {
+	const { commentId, previousStatus, refreshCommentListQuery: query, siteId, status } = action;
 
 	dispatch( removeNotice( `comment-notice-${ commentId }` ) );
 
+	if ( !! query && !! query.progressId ) {
```

Comment:
`get( query, 'progressId', null )`? corner case if we happen to use 0 of course.